dmoj/judge.py

```diff
@@ -458,21 +458,28 @@ def _grade_cases(self) -> Generator[Tuple[IPC, tuple], None, None]:
 
         flattened_cases: List[Tuple[Optional[int], Union[TestCase, BatchedTestCase]]] = []
         batch_number = 0
+        batch_requirements: List[Set[int]] = []
         for case in self.grader.cases():
             if isinstance(case, BatchedTestCase):
                 batch_number += 1
                 for batched_case in case.batched_cases:
                     flattened_cases.append((batch_number, batched_case))
+                batch_requirements.append(set(case.requirements))
             else:
                 flattened_cases.append((None, case))
 
         case_number = 0
         is_short_circuiting = False
         is_short_circuiting_enabled = self.submission.short_circuit
+        passed_batches: Set[int] = set()
         for batch_number, cases in groupby(flattened_cases, key=itemgetter(0)):
             if batch_number:
                 yield IPC.BATCH_BEGIN, (batch_number,)
 
+                reqs = batch_requirements[batch_number - 1]  # List is zero-indexed
```

Comment:
Nit: don't abbreviate words in variable names.

```suggestion
                requirements = batch_requirements[batch_number - 1]  # List is zero-indexed
```

Reply:
Fixed, also did this elsewhere (in the validation loop).